anaDACScan.py

```diff
@@ -0,0 +1,236 @@
+    parser.add_argument("--calFileList", type=str, help="File specifying which calFile to use for each OH. Format: OH1 filenameOH1 <newline> OH2 filenameOH2 <newline> etc", metavar="calFileList")
+    parser.add_argument('-o','--outfilename', dest='outfilename', type=str, default="DACFitData.root", help="Filename to which output information is written", metavar='outfilename')
+    parser.add_argument('--assignXErrors', dest='assignXErrors', action='store_true', help="Whether to assign errors for the X variable (which is actually plotted on the y-axis)")
+
+    args = parser.parse_args()
+
+    print("Analyzing: '%s'"%args.infilename)
+
+    # Set default histogram behavior
+    r.TH1.SetDefaultSumw2(False)
+    r.gROOT.SetBatch(True)
+    r.gStyle.SetOptStat(1111111)
+
+    dacScanFile = r.TFile(args.infilename)
+
+    import numpy as np
+    import root_numpy as rp
+    
+    list_bNames = ['link']
+    ohArray = rp.tree2array(tree=dacScanFile.dacScanTree, branches=list_bNames)
+    ohArray = np.unique(ohArray['link'])
+    
+    dataPath = os.getenv("DATA_PATH") 
+    
+    if len(args.infilename.split('/')) > 1 and len(args.infilename.split('/')[len(args.infilename.split('/')) - 2].split('.')) == 5:
+        scandate = args.infilename.split('/')[len(args.infilename.split('/')) - 2]
+    else:    
+        scandate = 'noscandate'
+
+    for oh in ohArray:
+        os.system("mkdir -p "+ dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate)
+            
+    # Determine which DAC was scanned and against which ADC
+    nameX = ""
+    nameY = ""
+    for event in dacScanFile.dacScanTree:
+        nameX = str(event.nameX.data())
+        nameY = str(event.nameY.data())
+        break # all entries will be the same
+
+    if nameY not in ['ADC0', 'ADC1']:
+        print("Error: unexpected value of nameY: '%s'"%nameY)
+        exit(1)
+
+    calInfo = {}
+
+    if args.calFileList != None:
+        for line in open(args.calFileList):
+            line = line.strip(' ').strip('\n')
+            first = line.split(' ')[0].strip(' ')
+            second = line.split(' ')[1].strip(' ')
+            tuple_calInfo = parseCalFile(second)
+            calInfo[int(first)] = {'slope' : tuple_calInfo[0], 'intercept' : tuple_calInfo[1]}
+
+    #for each OH, check if calibration files were provided, if not search for the calFile in the $DATA_PATH, if it is not there, then skip that OH for the rest of the script
+    for oh in ohArray:
+        if oh not in calInfo.keys():
+            calDacCalFile = "{0}/{1}/calFile_calDac_{1}.txt".format(dataPath,chamber_config[oh])
+            calDacCalFileExists = os.path.isfile(calDacCalFile)
+            if not calDacCalFileExists:
+                print("Skipping OH{0}, detector {1}, missing CFG_CAL_DAC Calibration file:\n\t{2}".format(
+                    oh,
+                    chamber_config[oh],
+                    calDacCalFile))
+                ohArray = np.delete(ohArray,(oh))
+
+
```

Comment:
unnecessary white space

Reply:
done

Reply:
still appear here, e.g. 178 & 179.  Perhaps change was not committed/pushed?

Reply:
sorry, fixed now